File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    answer = []
    reportdic = {}
    reportcount = {}
    for i in id_list:
        reportcount[i] = 0

    for i in range(len(report)):
        report[i] = report[i].split()
        if report[i][0] not in reportdic.keys():
            reportdic[report[i][0]] = [report[i][1]]
        else:
            reportdic[report[i][0]].append(report[i][1])
        reportcount[report[i][0]] = 0


    for i in reportdic.keys():
        reportdic[i] = list(set(reportdic[i]))
        for j in reportcount.keys():
            if(j in reportdic[i]):
                reportcount[j] += 1

    for i in id_list:
        count = 0
        if i in reportdic.keys():
            for j in reportdic[i]:
                if reportcount[j] >= k:
                    count +=1
        answer.append(count)
    return answer

File: 프로그래머스/lv1/92334. 신고 결과 받기/신고 결과 받기.js
function solution(id_list, report, k) {
    report = new Set(report)
    const answer = [],
          stop = [],
          user = {},
          report_user = {};
    for(val of id_list){
        user[val] = []
        report_user[val] = 0
    }
    
    for(val of report){
        const from_to = val.split(' ');
        user[from_to[0]].push(from_to[1])
        report_user[from_to[1]]+=1
        if(report_user[from_to[1]] >= k) stop.push(from_to[1]);
    }
    
    for(val in user){
        let count = 0;
        user[val].map((item) => {
            if(stop.includes(item)) count++;
        })
        answer.push(count)
    }
    return answer;
}